core/src/main/scala/cats/data/Kleisli.scala

```diff
@@ -257,6 +257,18 @@ sealed private[data] trait KleisliFunctions {
    */
   def local[M[_], A, R](f: R => R)(fa: Kleisli[M, R, A]): Kleisli[M, R, A] =
     Kleisli(r => fa.run(f(r)))
+
+  /**
+   * Lifts a function to a Kleisli.
+   * {{{
+   * scala> import cats.data.Kleisli
+   * scala> val stringify = Kleisli.fromFunction[Option, Int, String](_.toString)
+   * scala> stringify.run(42)
+   * res0: Option[String] = Some(42)
+   * }}}
+   */
+  def fromFunction[M[_]: Applicative, A, R](f: R => A): Kleisli[M, R, A] =
```

Comment:
Is this a candidate for some partial application tomfoolery so we might be able to say `Kleisli.fromFunction[Option]` and infer the `R` and `A`?

Reply:
it's a good idea, I will give it a closer look

Reply:
I guess we can't infer R anyway unless there's a type annotation for the whole thing. Probably the best we can do is `Kleisli.fromFunction[Option, Int]` and inferring the result.